bindings/nodejs/src/lib.rs

```diff
@@ -50,66 +78,189 @@ pub struct ObjectMeta {
             .stat()
             .await
             .map_err(format_napi_error)
             .unwrap();
 
-        let (secs, nsecs) = meta
-            .last_modified()
-            .map(|v| (v.unix_timestamp(), v.nanosecond()))
-            .unwrap_or((0, 0));
-
-        Ok(ObjectMeta {
-            location: path,
-            last_modified: DateTime::<Utc>::from_utc(
-                NaiveDateTime::from_timestamp_opt(secs, nsecs)
-                    .expect("returning timestamp must be valid"),
-                Utc,
-            ).timestamp(),
-            size: meta.content_length() as u32
-        })
+        exact_meta(meta)
     }
 
-    #[napi]
-    pub async fn write(&self,  path: String, content: Vec<u8>) -> Result<()> {
-        self.inner.object(&path)
-            .write(content)
+    #[napi(js_name="statSync")]
+    pub fn blocking_stat(&self) -> Result<ObjectMetadata> {
+        let meta = self.inner
+            .blocking_stat()
             .map_err(format_napi_error)
+            .unwrap();
+
+        exact_meta(meta)
+    }
+
+    #[napi]
+    pub async fn write(&self, content: Buffer) -> Result<()> {
+        let c = content.as_ref().to_owned();
+        self.inner
+            .write(c)
             .await
+            .map_err(format_napi_error)
+    }
+
+    #[napi(js_name="writeSync")]
+    pub fn blocking_write(&self, content: Buffer) -> Result<()> {
+        let c = content.as_ref().to_owned();
+        self.inner.blocking_write(c).map_err(format_napi_error)
     }
 
     #[napi]
-    pub async fn read(&self,  path: String) -> Result<Vec<u8>> {
-        let res = self.inner.object(&path)
+    pub async fn read(&self) -> Result<Buffer> {
+        let res = self.inner
             .read()
             .await
             .map_err(format_napi_error)
             .unwrap();
-        Ok(res)
+        Ok(res.into())
+    }
+
+    #[napi(js_name="readSync")]
+    #[napi]
+    pub fn blocking_read(&self) -> Result<Buffer> {
+        let res = self.inner
+            .blocking_read()
+            .map_err(format_napi_error)
+            .unwrap();
+        Ok(res.into())
+    }
+
+    #[napi]
+    pub async fn scan(&self) -> Result<ObjectLister> {
+        Ok(ObjectLister {
+            inner: self.inner
+                .scan()
+                .await
+                .map_err(format_napi_error)
+                .unwrap()
+        })
     }
 
     #[napi]
-    pub async fn delete(&self, path: String) -> Result<()> {
-        let o = self.inner.object(&path);
-        o.delete()
+    pub async fn delete(&self) -> Result<()> {
+        self.inner
+            .delete()
             .await
             .map_err(format_napi_error)
     }
+
+    #[napi(js_name="deleteSync")]
+    pub fn blocking_delete(&self) -> Result<()> {
+        self.inner
+            .blocking_delete()
+            .map_err(format_napi_error)
+    }
+}
+
+fn exact_meta(meta: opendal::ObjectMetadata) -> Result<ObjectMetadata> {
```

Comment:
I perfer to have an `ObjectMetadata(opendal::ObjectMetadata)` and provide API on it instead of exact them at once.